instrumentor/controllers/instrumentationdevice/common.go

```diff
@@ -157,46 +148,46 @@ func removeInstrumentationDeviceFromWorkload(ctx context.Context, kubeClient cli
 		return errors.New("unknown kind")
 	}
 
-	err := kubeClient.Get(ctx, client.ObjectKey{
-		Namespace: namespace,
-		Name:      workloadName,
-	}, workloadObj)
-	if err != nil {
-		return client.IgnoreNotFound(err)
-	}
-
-	result, err := controllerutil.CreateOrPatch(ctx, kubeClient, workloadObj, func() error {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
According to https://pkg.go.dev/k8s.io/client-go/util/retry#pkg-variables

> DefaultBackoff is the recommended backoff for a conflict where a client may be attempting to make an unrelated modification to a resource under active management by one or more controllers.

> DefaultRetry is the recommended retry for a conflict where multiple clients are making changes to the same resource.

So maybe `DefaultBackoff` is more fitting? I'm not sure